plt_darwin.go

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE.txt file in the project root for full license information.
+//-----------------------------------------------------------------------------
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"net"
+
+	"github.com/nsf/termbox-go"
+	"golang.org/x/sys/unix"
+)
+
+func getNetDevStats(stats *ethrNetStat) {
+	ifs, err := net.Interfaces()
+	if err != nil {
+		ui.printErr("%v", err)
+		return
+	}
+
+	for _, iface := range ifs {
+		if iface.Flags&net.FlagUp == 0 {
+			continue
+		}
+
+		ifaceData, err := getIfaceData(iface.Index)
+		if err != nil {
+			ui.printErr("failed to load data for interface %q: %v", iface.Name, err)
+			continue
+		}
+
+		stats.netDevStats = append(stats.netDevStats, ethrNetDevStat{
+			interfaceName: iface.Name,
+			rxBytes:       ifaceData.Data.Ibytes,
+			rxPkts:        ifaceData.Data.Ipackets,
+			txBytes:       ifaceData.Data.Obytes,
+			txPkts:        ifaceData.Data.Opackets,
+		})
+	}
+}
+
+func getTcpStats(stats *ethrNetStat) {
+}
+
+func hideCursor() {
+	termbox.SetCursor(0, 0)
+}
+
+func blockWindowResize() {
+}
+
+func getIfaceData(index int) (*ifMsghdr2, error) {
+	var data ifMsghdr2
+	rawData, err := unix.SysctlRaw("net", unix.AF_ROUTE, 0, 0, unix.NET_RT_IFLIST2, index)
+	if err != nil {
+		return nil, err
+	}
+	err = binary.Read(bytes.NewReader(rawData), binary.LittleEndian, &data)
+	return &data, err
+}
+
+type ifMsghdr2 struct {
```

Comment:
In order to avoid having ifMsghdr2 defined here, we could check with the Go team if they'd accept a patch adding it to golang.org/x/sys:

```
% git diff
diff --git a/unix/types_darwin.go b/unix/types_darwin.go
index 9fd2aaa..dc13c20 100644
--- a/unix/types_darwin.go
+++ b/unix/types_darwin.go
@@ -198,7 +198,9 @@ type FdSet C.fd_set

 const (
        SizeofIfMsghdr    = C.sizeof_struct_if_msghdr
+       SizeofIfMsghdr2   = C.sizeof_struct_if_msghdr2
        SizeofIfData      = C.sizeof_struct_if_data
+       SizeofIfData64    = C.sizeof_struct_if_data64
        SizeofIfaMsghdr   = C.sizeof_struct_ifa_msghdr
        SizeofIfmaMsghdr  = C.sizeof_struct_ifma_msghdr
        SizeofIfmaMsghdr2 = C.sizeof_struct_ifma_msghdr2
@@ -208,8 +210,12 @@ const (

 type IfMsghdr C.struct_if_msghdr

+type IfMsghdr2 C.struct_if_msghdr2
+
 type IfData C.struct_if_data

+type IfData64 C.struct_if_data64
+
 type IfaMsghdr C.struct_ifa_msghdr

 type IfmaMsghdr C.struct_ifma_msghdr
```

(then run ``GOOS=darwin GOARCH=amd64 ./mkall.sh``)